Makefile

```diff
@@ -302,6 +305,96 @@ update-mocks/%: %
 unittest/%: %
 	$(VALGRIND) $(VALGRIND_TEST_ARGS) $*
 
+# Installation directories
+prefix = /usr/local
+exec_prefix = $(prefix)
+bindir = $(exec_prefix)/bin
+libexecdir = $(exec_prefix)/libexec
+datadir = $(prefix)/share
+docdir = $(datadir)/doc/$(PKGNAME)
+mandir = $(datadir)/man
+man1dir = $(mandir)/man1
+man7dir = $(mandir)/man7
+
+# Commands
+MKDIR_P = mkdir -p
+INSTALL = install
+INSTALL_PROGRAM = $(INSTALL)
+INSTALL_DATA = $(INSTALL) -m 644
+
+# Tags needed by some package systems.
+PRE_INSTALL = :
+NORMAL_INSTALL = :
+POST_INSTALL = :
+PRE_UNINSTALL = :
+NORMAL_UNINSTALL = :
+POST_UNINSTALL = :
+
+# Target to create directories.
+installdirs:
+	@$(NORMAL_INSTALL)
+	$(MKDIR_P) $(DESTDIR)$(bindir)
+	$(MKDIR_P) $(DESTDIR)$(man1dir)
+	$(MKDIR_P) $(DESTDIR)$(man7dir)
+	$(MKDIR_P) $(DESTDIR)$(docdir)
+
+# Programs to install in bindir.
+# Note that subdaemons should properly be put in libexecdir,
+# but current lightningd expects them to be in the same
+# directory as itself.
+# This could be a problem later, if lightningd changes its
+# search path for subdaemons; it might pick up older versions
+# of the subdaemons, if a newer version is installed, without
+# uninstalling the previous version.
```

Comment:
I'd put a FIXME here, which is what I grep for to see what needs doing.  It should be trivial to fix at the end of this patch series, AFAICT...